### PR TITLE
*: fix improperly wrapped errors

### PR DESCRIPTION
**Pull request by rafiss**
refs #42510

I'm working on a linter that detects errors that are not wrapped
correctly, and it discovered these.

Release note: None

### CODE REVIEW

**Review by otan — APPROVED** [review]

**Review by RaduBerinde — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72353#-:-MnZSNE3Ei3kbzOsTn7h:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @RaduBerinde and @rafiss)

___
*[pkg/testutils/sqlutils/sql_runner.go, line 120 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/72353#-MnZS4TJ9Uc5m2yuTF5-:-MnZS4TJ9Uc5m2yuTF50:b-bkv20y) ([raw file](https://github.com/cockroachdb/cockroach/blob/634fd19257eec8c1a880c6c67fd95964f9f6424c/pkg/testutils/sqlutils/sql_runner.go#L120)):*
> ```Go
> 		_, err := sr.DB.ExecContext(context.Background(), query, args...)
> 		if !testutils.IsError(err, errRE) {
> 			return errors.AssertionFailedf("expected error '%s', got: %v", errRE, err)
> ```

[nit] I don't think this should be an AssertionFailed (we don't really care to see this stack). I would just extract `errStr := err.Error()` and use that with `Newf()`.

File: pkg/testutils/sqlutils/sql_runner.go [hunk 1]
```diff
@@ -117,7 +117,7 @@ func (sr *SQLRunner) ExpectErrSucceedsSoon(
 	sr.succeedsWithin(t, func() error {
 		_, err := sr.DB.ExecContext(context.Background(), query, args...)
 		if !testutils.IsError(err, errRE) {
-			return errors.Newf("expected error '%s', got: %v", errRE, err)
+			return errors.AssertionFailedf("expected error '%s', got: %v", errRE, err)
```
Comment (stevendanna) [inline pkg/testutils/sqlutils/sql_runner.go:120]: For my own understanding, why are we preferring AssertionFailedf in these cases?

**Review by stevendanna — APPROVED** [review]
Left two questions, but overall it looks good.

The `fmt.Errorf -> errors.Wrap` transformation isn't necessarily equivalent, but I checked all the ones here and they seem OK. I flagged one we might want to double-check.

**Review by rafiss — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72353)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @RaduBerinde, @rafiss, and @stevendanna)

___
*[pkg/cmd/roachprod-stress/main.go, line 313 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/72353#-MnXQtDiGQLeBI5KU3Kt-r1-313:-Mnb8F6c8O2HA8TkwFat:b-igr6oy) ([raw file](https://github.com/cockroachdb/cockroach/blob/634fd19257eec8c1a880c6c67fd95964f9f6424c/pkg/cmd/roachprod-stress/main.go#L313)):*
<details><summary><i>Previously, stevendanna (Steven Danna) wrote…</i></summary><blockquote>

It looks like before this would always return an error whereas this might now return nil.  However, I'm not sure it matters because I believe Err() only ever returns the two errors already caught in the switch above.
</blockquote></details>

more importantly:
```
	// If Done is closed, Err returns a non-nil error explaining why:
```
___
*[pkg/testutils/sqlutils/sql_runner.go, line 120 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/72353#-MnXQtDiGQLeBI5KU3Kz-r1-120:-Mnb8Slf5mjoxu_-Z0Ts:bjzv6jc) ([raw file](https://github.com/cockroachdb/cockroach/blob/634fd19257eec8c1a880c6c67fd95964f9f6424c/pkg/testutils/sqlutils/sql_runner.go#L120)):*
<details><summary><i>Previously, stevendanna (Steven Danna) wrote…</i></summary><blockquote>

For my own understanding, why are we preferring AssertionFailedf in these cases?
</blockquote></details>

my thinking was that it made sense since the test is asserting that the error is the expected type. but i agree with the point that the stack in this case is useless, so i'll change it back

actually, it's worth doing it explicitly anyway since `err.Error()` or `%v` doesn't show things like error details/hints, which might be useful to show here too.

**Comment by rafiss** [comment]
merging since failure is from a flaky test

bors r=otan,RaduBerinde,stevendanna

**Comment by rafiss** [comment]
bors r=otan,RaduBerinde,stevendanna

**Pull request merged**